### PR TITLE
Rebuild ethashproof

### PR DESCRIPTION
**Pull request by clearloop**
## In progress
+ [x] delete useless cache binary
+ [x] delete build.sh
+ [x] add makefile
+ [x] cross-compliation
+ [x] rename ethashproof to core
+ [ ] replace ethutils to local fetcher
+ [x] add ci

## Note

Should update `darwinia.js` after this merge.

### CODE REVIEW

File: build.sh [hunk 1]
```diff
@@ -1,6 +0,0 @@
-cd cmd/epoch
-go build -v
-cd ../relayer
-go build -v
-cd ../cache
-go build -v
```
Comment (hackfisher) [inline build.sh]: Could we keep the build.sh, it is easy for developers to build and use.

**Review by hackfisher — APPROVED** [review]

File: build.sh (hunk 1 above)
Comment (clearloop) [reply build.sh]: I think the `build.sh`'s code is not good...and `makefile` has the same function, for example:

Just run `make` under `/ethashproof` will build binaries to `target/`
+ target/relayer
+ target/cache
+ target/epoch

**Comment by clearloop** [comment]
Just find that we might don't have to fork this repo, just ref it and build our own commands is possible.

**Comment by clearloop** [comment]
The `ethash` package is not required in this repo actually, the core thing of this repo is the `mtree` package, ~~wondering if there is any repo implementing this more gently(maybe in `geth` or `openethereum`)~~

**Pull request closed without merging**